### PR TITLE
fix: dashboard search across folders

### LINKED ISSUE #5198: Enable searching dashboards across different folders
## Which OpenObserve functionalities are relevant/related to the feature request?

dashboards

## Description

We want to enable the user to search for dashboards by a case-insensitive title substring across all folders owned by the organization.

### Step 1 - backend - DONE ✅ 

Migrate folders and dashboards out of the `meta` table and into their own separate tables so that we can more efficiently search on the dashboard title, a field which is currently stored in JSON blobs in the `meta` table's `value` column.

### Step 2 - backend - DONE ✅ 

Add a `title` query parameter to the `ListDashboards` endpoint that will filter results by title.

### Step 3 - backend - DONE ✅ 

Add `folder_id` and `folder_name` fields to each JSON object returned by the `ListDashboards` endpoint. The frontend will need to be able to show these fields in the UI for cross-folder dashboard search.

### Step 4 - frontend - Est: ?

The UI needs to be updated to utilize the new `title` query parameter in the `ListDashboards` endpoint. It will also need to render information about each dashboard's parent folder in results from the cross-folder dashboard search.

### PR DESCRIPTION
**Pull request by nikhilsaikethe**
This PR fixes #5198 

## Summary by CodeRabbit

- **New Features**
	- Enhanced search functionality with a toggle to search across folders or within the current folder.
	- New methods for fetching search results and debouncing input for efficient searching.
	- Added support for dynamic filtering of dashboards based on the search context.
	- New localization entries for improved user interface text related to dashboard searches.
	- Added methods for managing permissions and fetching dashboards within role management.
	- Enhanced report creation with improved handling of dashboard options and scheduling.

- **Bug Fixes**
	- Improved error handling for dashboard management functions to prevent runtime errors.

- **Chores**
	- Streamlined dashboard state updates after modifications to ensure consistency.
	- Added `data-test` attributes to components for better testability.

### CODE REVIEW

File: web/src/views/Dashboards/Dashboards.vue [hunk 1]
```diff
@@ -756,9 +892,17 @@
...
       selectedDashboardIdToMove,
       showMoveDashboardDialog,
       handleDashboardMoved,
+      searchAcrossFolders,
+      searchQuery,
+      fetchSearchResults,
+      debouncedSearch,
+      filteredResults,
+      activeFolderToMove,
+      hasSearched,
```
[the remark this answers is not in the document]
Comment (kethesainikhil) [reply web/src/views/Dashboards/Dashboards.vue:901]: yes this has been removed

[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/views/Dashboards/Dashboards.vue:901]: yes this has been removed

File: web/src/views/Dashboards/Dashboards.vue [hunk 2]
```diff
@@ -702,6 +789,55 @@
...
+          );
+          return migratedDashboards;
+      } catch (error) {
+        q.notify({
+          message: "Error fetching search results",
+          color: "negative",
+        });
+      } finally {
+        loading.value = false;
+      }
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/views/Dashboards/Dashboards.vue]: yes considered this suggestion

File: web/src/views/Dashboards/Dashboards.vue [hunk 3]
```diff
@@ -508,6 +548,18 @@
...
+      await debouncedSearch(newQuery);
+    });
+
+    watch(searchAcrossFolders, async (newVal) => {
+      if (newVal) {
+        searchQuery.value = "";
+        filteredResults.value = [];
+        hasSearched.value = false;
+      }
+    });
```
[the remark this answers is not in the document]
Comment (ktx-kirtan) [reply web/src/views/Dashboards/Dashboards.vue:561]: It will not throw an error.

**Review by ktx-kirtan — APPROVED** [review]

**Pull request merged**